### ISSUE TITLE
equals issue?

### ISSUE THREAD

**Issue by dizzzz** [eXist-db/exist#915]
in org.exist.storage.NativeValueIndex.ExactMatcher#matches the following code is there:

```java
    private static class ExactMatcher implements TermMatcher
    {
        private String expr;

        public ExactMatcher( String expr ) throws EXistException
        {
            this.expr = expr;
        }

        public boolean matches( CharSequence term )
        {
            return( term.toString() == expr );
        }
    }
```

intellij gives a warning for `term.toString() == expr` 

```
Name      public method boolean matches(CharSequence term)   
Location      class ExactMatcher (org.exist.storage.NativeValueIndex)    
Problem synopsis      String values are compared using '==', not 'equals()' at line 1384
```

is IntelliJ correct here? Or is an object comparison really sufficient? @wolfgangmm

**Comment by shabanovd** [comment]
That one looks as bug for me.

**Comment by adamretter** [comment]
Yes

**Comment by dizzzz** [comment]
Yes..... Means???

**Comment by adamretter** [comment]
Yes intellij is correct. It is s bug, you should never compare strings by reference

**Comment by dizzzz** [comment]
Ok I'll fix

**Comment by dizzzz** [comment]
fixes #915
